src/mtcp/mtcp_restart.c

```diff
@@ -463,11 +463,25 @@ static void restart_slow_path()
   restorememoryareas(&rinfo);
 }
 
-static void mtcp_simulateread(int fd)
+static void mtcp_simulateread(int fd, MtcpHeader *mtcpHdr)
 {
   int mtcp_sys_errno;
+
+  // Print miscellaneous information:
+  char buf[MTCP_SIGNATURE_LEN+1];
+  mtcp_memcpy(buf, mtcpHdr->signature, MTCP_SIGNATURE_LEN);
+  buf[MTCP_SIGNATURE_LEN] = '\0';
+  mtcp_printf("\nMTCP: %s", buf);
```

Comment:
The primary use of this function is to print to stdout when called by readdmtcp.sh.  The routine MTCP_PRINTF() will print the file, function, line number, etc., followed by an indented message on the next line.  I don't think people calling readdmtcp.sh would want to see what function and line number is delivering the information to them.  (For example, readelf doesn't do that.)